cmd/cli/devstack/devstack.go

```diff
@@ -206,12 +207,39 @@ func runDevstack(cmd *cobra.Command, cfg types.BacalhauConfig, fsr *repo.FsRepo,
 		devstack.WithComputeConfig(computeConfig),
 		devstack.WithRequesterConfig(requesterConfig),
 	)
+	userKeyPath, err := fsr.UserKeyPath()
+	if err != nil {
+		return err
+	}
+	userKey, err := baccrypto.LoadUserKey(userKeyPath)
+	if err != nil {
+		return err
+	}
+	executionDir, err := fsr.ExecutionDir()
+	if err != nil {
+		return err
+	}
 	if IsNoop {
 		options = append(options, devstack.WithDependencyInjector(devstack.NewNoopNodeDependencyInjector()))
 	} else if ODs.ExecutorPlugins {
-		options = append(options, devstack.WithDependencyInjector(node.NewExecutorPluginNodeDependencyInjector(cfg)))
+		pluginPath, err := fsr.EnginePluginsDir()
+		if err != nil {
+			return err
+		}
+		options = append(options, devstack.WithDependencyInjector(
```

Comment:
every node in the devstack should have its own config and DataDir. Though I understand that might be a bigger change for this PR

Reply:
I agree with both points. Changing this behavior in the devstack is beyond the scope of this PR.